src/main/java/appeng/tile/powersink/AEBasePoweredTile.java

```diff
@@ -19,7 +19,308 @@
+	private AccessRestriction internalPowerFlow = AccessRestriction.READ_WRITE;
+	// the current power buffer.
+	private double internalCurrentPower = 0;
+	private EnumSet<EnumFacing> internalPowerSides = EnumSet.allOf( EnumFacing.class );
+	private final IEnergyStorage forgeEnergyAdapter;
+	private Object teslaEnergyAdapter;
+
+	private IC2PowerSink ic2Sink;
+
+	public AEBasePoweredTile()
+	{
+		this.forgeEnergyAdapter = new ForgeEnergyAdapter( this );
+		if( Capabilities.TESLA_CONSUMER != null )
+		{
+			this.teslaEnergyAdapter = new TeslaEnergyAdapter( this );
+		}
+		this.ic2Sink = Integrations.ic2().createPowerSink( this, this );
+		this.ic2Sink.setValidFaces( this.internalPowerSides );
+	}
+
+	protected EnumSet<EnumFacing> getPowerSides()
+	{
+		return this.internalPowerSides.clone();
+	}
+
+	protected void setPowerSides( final EnumSet<EnumFacing> sides )
+	{
+		this.internalPowerSides = sides;
+		this.ic2Sink.setValidFaces( sides );
+		// trigger re-calc!
+	}
+
+	@TileEvent( TileEventType.WORLD_NBT_WRITE )
+	public void writeToNBT_AERootPoweredTile( final NBTTagCompound data )
+	{
+		data.setDouble( "internalCurrentPower", this.getInternalCurrentPower() );
+	}
+
+	@TileEvent( TileEventType.WORLD_NBT_READ )
+	public void readFromNBT_AERootPoweredTile( final NBTTagCompound data )
+	{
+		this.setInternalCurrentPower( data.getDouble( "internalCurrentPower" ) );
+	}
+
+	@Override
+	public final double getExternalPowerDemand( final PowerUnits externalUnit, final double maxPowerRequired )
+	{
+		return PowerUnits.AE.convertTo( externalUnit, Math.max( 0.0, this.getFunnelPowerDemand( externalUnit.convertTo( PowerUnits.AE, maxPowerRequired ) ) ) );
+	}
+
+	protected double getFunnelPowerDemand( final double maxRequired )
+	{
+		return this.getInternalMaxPower() - this.getInternalCurrentPower();
+	}
+
+	@Override
+	public final double injectExternalPower( final PowerUnits input, final double amt )
+	{
+		return PowerUnits.AE.convertTo( input, this.funnelPowerIntoStorage( input.convertTo( PowerUnits.AE, amt ), Actionable.MODULATE ) );
+	}
+
+	protected double funnelPowerIntoStorage( final double power, final Actionable mode )
+	{
+		return this.injectAEPower( power, mode );
+	}
+
+	@Override
+	public final double injectAEPower( double amt, final Actionable mode )
+	{
+		if( amt < 0.000001 )
+		{
+			return 0;
+		}
+
+		if( mode == Actionable.SIMULATE )
+		{
+			final double fakeBattery = this.getInternalCurrentPower() + amt;
+
+			if( fakeBattery > this.getInternalMaxPower() )
+			{
+				return fakeBattery - this.getInternalMaxPower();
+			}
+
+			return 0;
+		}
+		else
+		{
+			if( this.getInternalCurrentPower() < 0.01 && amt > 0.01 )
+			{
+				this.PowerEvent( PowerEventType.PROVIDE_POWER );
+			}
+
+			this.setInternalCurrentPower( this.getInternalCurrentPower() + amt );
+			if( this.getInternalCurrentPower() > this.getInternalMaxPower() )
+			{
+				amt = this.getInternalCurrentPower() - this.getInternalMaxPower();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "amt". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1226)